src/py/flwr/server/driver/inmemory_driver.py

```diff
@@ -146,7 +178,10 @@ def pull_messages(self, message_ids: Iterable[str]) -> Iterable[Message]:
         # Delete
         self.state.delete_messages(message_ins_ids=message_ins_ids_to_delete)
 
-        return message_res_list
+        def iter_msg() -> Iterator[Message]:
+            return iter(message_res_list)
+
+        return iter_msg()
```

Comment:
Both `Iterator` and `list` are acceptable types of `Iterable`. Since we already have a list of messages, we don't need to convert it to an iterator here.

@jafermarq Should we also update the in memory driver so it really returns an iterator and doesn't pull all messages at once? Wdyt?

```suggestion
        return message_res_list
```

Reply:
> Should we also update the in memory driver so it really returns an iterator and doesn't pull all messages at once? Wdyt?

Indeed, that was the original idea - to keep the return behaviour the same for both in-memory driver and the `grpc_driver`.